### PR TITLE
ENH: Add pydantic meta schema

### PR DESCRIPTION
**Pull request by janbjorge**
Addresses #423 

This PR is does not change any logic, it just adds a Pydantic model to the repository. There a few convenience tools as well, that i used while dev. the pydantic model.

### CODE REVIEW

**Comment by janbjorge** [comment]
In its current state it fully runs boom.py without any validation errors.

File: pyproject.toml [hunk 1]
```diff
@@ -35,6 +35,7 @@ dependencies = [
     "numpy",
     "pandas",
     "pyarrow",
+    "pydantic>=2.5.0",
```
Comment (janbjorge) [inline pyproject.toml:38]: Is this version right?

**Comment by perolavsvendsen** [comment]
- [X] Verify Pydantic version with Komodo and RMS environments. _We think this should be OK._
- [X] Existing tests should pass with new schema.
- [ ] Tests should be expanded if coverage is too poor. _Scope for this PR??_
- [ ] Existing data should validate on the new schema. _Scope for this PR??_
- [X] Add more elaborate docstrings on the data classes, for documentation (combine with checking Sphinx capabilities to avoid double work). _Later PR, not scope._
- [ ] Check which "volume" variants in the content list that are actually discontinued.
- [X] Need to be more specific on some of the numerical inputs, e.g. cannot have negative number of layers etc. _Later PR, not scope._
- [X] Build the schema to the Radix app. _Later PR, not scope._

File: src/fmu/dataio/models/meta/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from .model import Root, dump
```
Comment (perolavsvendsen) [inline src/fmu/dataio/models/meta/__init__.py:1]: May cause confusion when the word "model" hits the namespace in a context where this word has lots of meanings. Possible to use "datamodels"?

**Comment by janbjorge** [comment]
From review, rename _modeles_ folder to _datastructure_

File: src/fmu/dataio/models/meta/__init__.py (hunk 2 above)
Comment (perolavsvendsen) [reply src/fmu/dataio/models/meta/__init__.py:1]: "datastructure" or "datastructures"? Ref discussion

File: pyproject.toml (hunk 1 above)
Comment (janbjorge) [reply pyproject.toml:38]: Checked, this is fine.

**Review by jcrivenaes — APPROVED** [review]
🚀

File: src/fmu/dataio/models/meta/content.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+class PinchoutContent(Content):
+    content: Literal[enums.ContentEnum.pinchout]
+
+
+class PropertyContent(Content):
+    content: Literal[enums.ContentEnum.property]
+
+
+class PTVContent(Content):
+    content: Literal[enums.ContentEnum.pvt]
```
[the remark this answers is not in the document]
Comment (janbjorge) [reply src/fmu/dataio/models/meta/content.py]: Should be pvt, thanks.

**Comment by janbjorge** [comment]
Followups:

- It seems that SUMO will add timezone if not given, leads to unnecessary union of aware and naive datetime (se code comments).
- RealizationJobs is to strict, or sumo is lacking information in old cases. Needs followup.

```python
python3 tools/sumo-explorer-model-validate.py
Traceback (most recent call last):
  File "/Users/JLOV/src/fmu-dataio/tools/sumo-explorer-model-validate.py", line 53, in <module>
    parsed = Root.model_validate(m)
  File "/Users/JLOV/.pyenv/versions/dataio/lib/python3.10/site-packages/pydantic/main.py", line 503, in model_validate
    return cls.__pydantic_validator__.validate_python(
pydantic_core._pydantic_core.ValidationError: 16 validation errors for Root
cube.fmu.realization.jobs.ert_pid
  Field required [type=missing, input_value={'umask': '0002', 'DATA_R...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.global_environment
  Field required [type=missing, input_value={'umask': '0002', 'DATA_R...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.arg_types
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.argList
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.error_file
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.executable
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.license_path
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.max_arg
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.max_running_minutes
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.max_running
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.start_file
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.stderr
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.stdin
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.stdout
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.jobList.0.target_file
  Field required [type=missing, input_value={'min_arg': 1, 'name': 'M...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
cube.fmu.realization.jobs.run_id
  Field required [type=missing, input_value={'umask': '0002', 'DATA_R...'anything': 'something'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.5/v/missing
```

**Pull request merged**